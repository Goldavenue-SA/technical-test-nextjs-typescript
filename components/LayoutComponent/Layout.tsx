import type { ReactElement } from 'react';
import { Background, Container, Main, Menu, OverlayPattern } from './Styles';
import Image from 'next/image';
import Link from 'next/link';

export const Layout = (page: ReactElement): JSX.Element => {
  return (
    <>
      <Background>
        <OverlayPattern />
        <Menu>
          <Link href={'/'} aria-label="Go to home page">
            <Image
              src="/images/pokeball.png"
              alt="Pokeball Image"
              width={96}
              height={96}
            />
            <span>Home</span>
          </Link>
        </Menu>
        <Container>
          <Main>{page}</Main>
        </Container>
      </Background>
    </>
  );
};
