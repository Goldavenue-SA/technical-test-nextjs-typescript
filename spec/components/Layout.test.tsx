import { render } from '@testing-library/react';
import type { ReactElement } from 'react';
import { Layout } from '../../components/LayoutComponent/Layout';

describe('Test Layout Component', () => {
  it('should match snapshot with a mock page', () => {
    const mockPage: ReactElement = <div>Mock Page Content</div>;

    const { asFragment } = render(<>{Layout(mockPage)}</>);

    expect(asFragment()).toMatchSnapshot();
  });
});
