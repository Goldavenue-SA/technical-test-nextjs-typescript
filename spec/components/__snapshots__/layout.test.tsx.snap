// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout Component should match snapshot with a mock page 1`] = `
<DocumentFragment>
  <div
    class="sc-f8e2d048-0 bQCGGJ"
  >
    <div
      class="sc-f8e2d048-1 lbvEgI"
    />
    <nav
      class="sc-f8e2d048-2 fPbujT"
    >
      <a
        aria-label="Go to home page"
        href="/"
      >
        <img
          alt="Pokeball Image"
          data-nimg="1"
          decoding="async"
          height="96"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fpokeball.png&w=256&q=75"
          srcset="/_next/image?url=%2Fimages%2Fpokeball.png&w=96&q=75 1x, /_next/image?url=%2Fimages%2Fpokeball.png&w=256&q=75 2x"
          style="color: transparent;"
          width="96"
        />
        <span>
          Home
        </span>
      </a>
    </nav>
    <div
      class="sc-f8e2d048-3 hYhOah"
    >
      <div
        class="sc-f8e2d048-4 gSEfTz"
      >
        <div>
          Mock Page Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
